### PR TITLE
GODRIVER-1405 set appname on ServerConfig

### PR DESCRIPTION
**Pull request by iwysiu**

### CODE REVIEW

File: x/mongo/driver/topology/server_options.go [hunk 1]
```diff
@@ -68,6 +68,14 @@ func WithCompressionOptions(fn func(...string) []string) ServerOption {
 	}
 }
 
+// WithServerAppName configures the server's application name.
+func WithServerAppName(fn func(string) string) ServerOption {
```
Comment (jyemin) [inline x/mongo/driver/topology/server_options.go]: Can this be unit tested?

File: x/mongo/driver/topology/topology_options.go [hunk 2]
```diff
@@ -64,7 +64,8 @@ func WithConnString(fn func(connstring.ConnString) connstring.ConnString) Option
 		var connOpts []ConnectionOption
 
 		if cs.AppName != "" {
-			connOpts = append(connOpts, WithAppName(func(string) string { return cs.AppName }))
+			connOpts = append(connOpts, WithConnectionAppName(func(string) string { return cs.AppName }))
+			c.serverOpts = append(c.serverOpts, WithServerAppName(func(string) string { return cs.AppName }))
```
Comment (jyemin) [inline x/mongo/driver/topology/topology_options.go]: Can this be unit tested?

File: x/mongo/driver/topology/server_options.go (hunk 1 above)
Comment (iwysiu) [reply x/mongo/driver/topology/server_options.go]: Done

File: x/mongo/driver/topology/topology_options.go (hunk 2 above)
Comment (iwysiu) [reply x/mongo/driver/topology/topology_options.go]: Done

**Pull request merged**